Add Visibility component documentation (#5326)
mintlify/docs: ai-native.mdx

### Discovering

> Your site is automatically optimized for AI tools and search engines to help users discover your documentation. All pages send their content as Markdown to AI agents instead of HTML, which helps these tools process your content faster and use fewer tokens. Every page is also available to view as Markdown by appending `.md` to the URL. Use the [`Visibility`](/components/visibility) component to tailor content for each audience—show UI instructions to human readers and API details to AI agents.

Reviewer: this could be tweaked a bit - as "show UI instructions to human readers and API details to AI agents" doesn't strongly capture the goal of this component and is too specific. we should make this more generic, something like:
```suggestion
Your site is automatically optimized for AI tools and search engines to help users discover your documentation. All pages send their content as Markdown to AI agents instead of HTML, which helps these tools process your content faster and use fewer tokens. Every page is also available to view as Markdown by appending `.md` to the URL. Use the [`Visibility`](/components/visibility) component to tailor content for each audience—determine what content is shown only in the web UI and what content is only included in markdown.
```

Reviewer 2: ```suggestion
Your site is automatically optimized for AI tools and search engines to help users discover your documentation. All pages send their content as Markdown to AI agents instead of HTML, which helps these tools process your content faster and use fewer tokens. Every page is also available to view as Markdown by appending `.md` to the URL. Use the [visibility](/components/visibility) component to tailor content for each audience.
```

I think we can just drop the second part of the sentence since this is more of an overview page and links people out for more info. "Tailor to each audience" is enough for people to go read more if they want to do that in my opinion.
